Sources/Experiment/ExperimentPlugin.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  Excperiment.swift
+//  experiment-ios-client
+//
+//  Created by Chris Leonavicius on 3/18/25.
+//
+
+import AmplitudeCore
+import Foundation
+
+@objc
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public class ExperimentPlugin: NSObject, UniversalPlugin {
+
+    @objcMembers
+    public class Config : NSObject {
+
+        public let deploymentKey: String?
+        public let debug: Bool
+        public let fallbackVariant: Variant
+        public let initialFlags: String?
+        public let initialVariants: [String: Variant]
+        public let source: Source
+        public let serverUrl: String
+        public let flagsServerUrl: String
+        public let fetchTimeoutMillis: Int
+        public let retryFetchOnFailure: Bool
+        public let automaticExposureTracking: Bool
+        public let fetchOnStart: NSNumber? // objc cant do nil boolean values, use nsnumber
+        public let pollOnStart: Bool
+        public let flagConfigPollingIntervalMillis: Int
+        public let automaticFetchOnAmplitudeIdentityChange: Bool
+
+        public init(deploymentKey: String? = nil,
```

Comment:
Any particular reason we dont just pass in the config object?

Reply:
Yes, there are options (`instanceName`, `serverZone`, `userProvider`, `exposureTrackingProvider`) which should not be externally configured when using this plugin.

If we keep the standard config object, what is the expected behavior when a user tries to add these properties?